stubs/phantom/base_connector.py

```diff
@@ -25,10 +33,22 @@ def load_state(self):
         pass
 
     def get_config(self):
-        pass
+        return self._config
 
     def save_state(self, state):
-        pass
+        if not state:
+            return
+        if not self._state:
+            self._state = {**state}
+        else:
+            self._state = {**state, **self._state}
```

Comment:
The order of these means the existing state values will override the new values, which isn't what we want, for example:

```
>>> d1 = {"a": 1, "b": 2}
>>> d2 = {"a": 5, "c": 5}
>>> 
>>> {**d1, **d2}
{'a': 5, 'b': 2, 'c': 5}
>>> {**d2, **d1}
{'a': 1, 'c': 5, 'b': 2}
```

We should probably just do `self._state.update(state)`.